src/azul/debug.py

```diff
@@ -0,0 +1,67 @@
+from contextlib import (
+    ContextDecorator,
+)
+from logging import (
+    getLogger,
+)
+
+from azul import (
+    config,
+)
+
+log = getLogger(__name__)
+
+
+class RemoteDebugSession(ContextDecorator):
+    """
+    Adapted from https://stackoverflow.com/a/60333250/7830612
+
+    According to the Stackoverflow post, since the debug session is over a
+    long-running TCP connection, we need to close the connection explicitly
+    within the lambda, otherwise the lambda will time out. A context manager
+    handles this conveniently.
+
+    For usage instructions, see the README.
+    """
+
+    def __init__(self):
+        self.active = False
+        if self.is_available():
+            import pydevd
+            self.pydevd = pydevd
+        else:
+            self.pydevd = None
+            log.warning('Remote debugging is not available')
+
+    def __enter__(self):
+        if self.is_available():
```

Comment:
This should fail strictly so the we don't forget to remove calls to the CM.

Reply:
This is no longer relevant given some other refactoring I've done.